### DISCUSSION TITLE
[keyball44]テスト用ファームウェアでしか右キーボード(トラックボール側)が動作しない

### DISCUSSION THREAD

**Discussion by yuketenwalker** [Yowkees/keyball#630] [Q&A]
**使用モデル**：keyball44（右トラックボール）※USBは左に接続している

**【症状１】**
- 組み立てが完了し、左右に正規ファームウェアを書き込むと右側のキー、トラックボールすべて反応しなくなる
- pro microは認識されているが断続的に接続音(ﾋﾟﾛﾝ)が鳴り、接続・切断を繰り返す(※動画添付)
- テスト用ファームウェア確認時は右側も文字入力できていたので（ただし左のキーが反転した形で入力され、トラックボール進行方向が反転）右側だけテスト用ファームウェアに戻すと動作するようになる（キーも正常に入力され、トラックボール進行方向も正常）

**【症状２】**
- (上記の右側だけテストファームウェア入れた状態でも)右キーボードの右端1列はずっと入力できない

**【その他】**
当初LEDをはんだ付けしたが、テストファームウェアでの確認時に左pro maicroが一切反応せず。
過去の類似質問を調べ、LED回路がどこかでショートしている可能性が高いっぽかったので、
回答に記載されていたLED CUTと書いてあるジャンパ（LED回路を一括で切断できるとの事）を
カッターで削る方法を実施し、問題解消した。(写真添付）

初めての自作キーボード製作なので、原因が分からず手詰まりの状態です。
お知恵をお借りできると大変有難いです。
何卒、よろしくお願いいたします。

https://github.com/user-attachments/assets/ad18b3ba-453a-4ea5-ad0f-dd44ee6743f1

![20240724_132057](https://github.com/user-attachments/assets/8360787d-0175-4b72-b058-f466a9d07e3f)
![20240724_133307](https://github.com/user-attachments/assets/a8709718-65d1-4ce6-9843-0d3b38448160)

**Comment by Ryo624** [comment]
右手側のTRRSジャックのはんだが端子に上手く馴染んでいないように見えるのでそちらをやり直してみるとどうなりますでしょうか。

**Reply by yuketenwalker** [reply]
ご助言下さり、ありがとうございます。
TRRSジャックのだけではなく、ダイオードのはんだ付けやり直してみましたが

- 右キーボードのみテスト用ファームウェアを入れれば、トラックボール、キーは入力できる。(PC接続は左)
- PC接続を右にすると、キーもトラックボールも左右反転してしまう
- 右キーボードの右端1列は常に反応しない
- 右キーボードに正規ファームウェアを入れると接続、切断を繰り返し、入力できない（OLEDのロゴ表示も消える）


上記の状況は変わらなかったです。。
右側のはんだ付け含めた基盤の問題かなぁ～とは思うのですが、
白金ラボさんに問い合わせれば右側の基盤とはんだ付けしてしまっているパーツだけ、
再度購入したりって可能なのでしょうか？

お手数おかけしてしまい申し訳ありません。

![20240724_182839](https://github.com/user-attachments/assets/5f1147f3-b002-4167-a993-558c0b0f5934)

**Reply by Ryo624** [reply]
ProMicroの表側ははんだ付けされてますか？
ProMicroを左右で入れ替えるとどうなるかも確かめていただければ幸いです。

**Reply by yuketenwalker** [reply]
![20240725_150553](https://github.com/user-attachments/assets/30dcb0b5-f04f-41b7-8ad6-3ad2c62500ed)
ハンダ付けはできているかと思うのですが、どうでしょうか？

pro  micro入れ替え(PC接続は左)を行い、下記の結果になりました。
【左】pro microA(正規)　　　【右】pro microB(テスト用)　　←右キーボード入力可能
　　　　　　　　　　　↓
【左】pro microB(テスト用)　【右】pro microA(正規)　　　　←右キーボード反応なし

**Comment by Yowkees** [comment]
ご確認ありがとうございます。
左右判定ジャンパー周りやTRRSコネクタの半田付けは修正いただけたようですので、フルカラーLEDの信号線がProMicro動作に影響しているかも知れません。
LEDをすべて除去すると改善するかも知れませんが、確かに作り直してしまったほうが早いかもです。
ミドル基板と半田付け部品セットを購入者様向け価格で販売もできます。
発送先等ご相談したいので、お手数ですが下記お問い合わせよりご連絡いただければ幸いです。
https://shirogane-lab.net/inquiry
